### PR TITLE
Fix Octopart JS Error

### LINKED ISSUE #2218: Octopart: JS error
Seeing this error get thrown periodically, need to guard against it:
```
TypeError: null is not an object (evaluating 'item.avg_price_v2[0].toFixed')
```

------
IA Page:  http://duck.co/ia/view/octopart

### PR DESCRIPTION
**Pull request by SibuStephen**
Fixed #2218

### CODE REVIEW

**Comment by daxtheduck** [comment]
@SibuStephen Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please send an email to <quackslack@duckduckgo.com>.

>More Info: https://duck.co/duckduckhack/submission_and_review

Thanks!

**Comment by moollaza** [comment]
@SibuStephen I've updated the description slightly. It literally needs to say "**fixed** #2218" in order for GitHub to close the related issue when we merge the PR.

>More info: https://help.github.com/articles/closing-issues-via-commit-messages/, https://github.com/blog/1506-closing-issues-via-pull-requests

File: share/spice/octopart/octopart.js [hunk 1]
```diff
@@ -43,16 +43,20 @@
...
+                 
+                 if (!item.avg_price_v2[0]){
+                     return null;
+                 }
                 var images = get_images(item.imagesets),
                     datasheet = item.datasheets && item.datasheets[0];
 
                 return {
                     brand: item.brand.name,
-                    price: item.avg_price_v2[1] + ' $' + item.avg_price_v2[0].toFixed(2),
```
Comment (moollaza) [inline share/spice/octopart/octopart.js]: This is actually still required. If you take a look at the API result for some other queries, like https://duckduckgo.com/js/spice/octopart/atmega you'll see that the value is usually several decimal places, so we need this to ensure it's rounded properly.

Comment (SibuStephen) [inline share/spice/octopart/octopart.js]: Yes @moollaza  i got the point, `toFixed()` function is required for decimal values  upto 2 , will fix it right away. :+1:

File: share/spice/octopart/octopart.js [hunk 2]
```diff
@@ -43,14 +43,19 @@
                 sourceName: 'Octopart',
                 sourceUrl : 'http://octopart.com/search?q=' + api_result.request.q
             },
+         
             normalize: function(item) {
-
-                item = item.item; // our item object is wrapped in an "item" property
-
+               
+                var item = item.item;  // our item object is wrapped in an "item" property
```
Comment (moollaza) [inline share/spice/octopart/octopart.js]: Seeing as we're declaring multiple variables it makes sense to remove the 2nd `var`:

```js
var item = item.item, // ...
    images = ...,
    datasheet = ...;
```

File: share/spice/octopart/octopart.js [hunk 3]
```diff
@@ -43,14 +43,19 @@
...
                 var images = get_images(item.imagesets),
                     datasheet = item.datasheets && item.datasheets[0];
-
+                                 
+                if(!item.avg_price_v2[0]) {
+                    return null;
+                  }
```
Comment (moollaza) [inline share/spice/octopart/octopart.js]: nit: your indentation is off here:

```js
if (...) {
   ...
} else {
   ...
}
```

Comment (moollaza) [inline share/spice/octopart/octopart.js]: Actually, you don't even need the `else` block here.

File: share/spice/octopart/octopart.js (hunk 2 above)
Comment (SibuStephen) [inline share/spice/octopart/octopart.js]: @moollaza  Please take a look at the changes.

File: share/spice/octopart/octopart.js [hunk 4]
```diff
@@ -43,14 +43,18 @@
                 sourceName: 'Octopart',
                 sourceUrl : 'http://octopart.com/search?q=' + api_result.request.q
             },
+         
             normalize: function(item) {
-
-                item = item.item; // our item object is wrapped in an "item" property
-
+               
+                      item = item.item;  // our item object is wrapped in an "item" property
```
Comment (jagtalon) [inline share/spice/octopart/octopart.js]: Nitpick: Please indent this, i.e.,

```javascript
item = item.item;  // our item object is wrapped in an "item" property
var images = get_images(item.imagesets),
...
```

File: share/spice/octopart/octopart.js [hunk 5]
```diff
@@ -43,14 +43,18 @@
...
                 var images = get_images(item.imagesets),
                     datasheet = item.datasheets && item.datasheets[0];
-
+                                 
+                if(!item.avg_price_v2[0]) {
```
Comment (jagtalon) [inline share/spice/octopart/octopart.js]: Good! Although would something like `item.avg_price_v2.length > 0` be better?

Comment (SibuStephen) [inline share/spice/octopart/octopart.js]: @jagtalon Please check the indentation.

**Comment by jagtalon** [comment]
:+1:

**Pull request merged**